src/assignment/views/AssignmentDetail.tsx

```diff
@@ -0,0 +1,291 @@
+	user,
+	history,
+}) => {
+	const { tText, tHtml } = useTranslation();
+
+	const [assignmentLoading, setAssigmentLoading] = useState(false);
+	const [assignmentError, setAssigmentError] = useState<Partial<ErrorViewQueryParams> | null>(
+		null
+	);
+	const [assignment, setAssignment] = useAssignmentForm(undefined);
+
+	const id = match.params.id;
+
+	const fetchAssignment = useCallback(async () => {
+		try {
+			setAssigmentLoading(true);
+			setAssigmentError(null);
+
+			let tempAssignment: Assignment_v2_With_Blocks | null = null;
+
+			try {
+				tempAssignment = await AssignmentService.fetchAssignmentById(id);
+			} catch (err) {
+				if (JSON.stringify(err).includes(NO_RIGHTS_ERROR_MESSAGE)) {
+					setAssigmentError({
+						message: tHtml(
+							'assignment/views/assignment-edit___je-hebt-geen-rechten-om-deze-opdracht-te-bewerken'
+						),
+						icon: IconName.lock,
+						actionButtons: ['home'],
+					});
+					setAssigmentLoading(false);
+					return;
+				}
+				setAssigmentError({
+					message: tHtml(
+						'assignment/views/assignment-edit___het-ophalen-van-de-opdracht-is-mislukt'
+					),
+					icon: IconName.alertTriangle,
+					actionButtons: ['home'],
+				});
+				setAssigmentLoading(false);
+				return;
+			}
+
+			if (!tempAssignment) {
+				setAssigmentError({
+					message: tHtml(
+						'assignment/views/assignment-edit___het-ophalen-van-de-opdracht-is-mislukt'
+					),
+					icon: IconName.alertTriangle,
+					actionButtons: ['home'],
+				});
+				setAssigmentLoading(false);
+				return;
+			}
+
+			if (
+				!(await PermissionService.hasPermissions(
+					[
+						PermissionName.EDIT_ANY_ASSIGNMENTS,
+						{ name: PermissionName.EDIT_OWN_ASSIGNMENTS, obj: tempAssignment },
+					],
+					user
+				))
+			) {
+				setAssigmentError({
+					message: tHtml(
+						'assignment/views/assignment-edit___je-hebt-geen-rechten-om-deze-opdracht-te-bewerken'
+					),
+					icon: IconName.lock,
+					actionButtons: ['home'],
+				});
+				setAssigmentLoading(false);
+				return;
+			}
+
+			setAssignment(tempAssignment);
+		} catch (err) {
+			setAssigmentError({
+				message: tHtml(
+					'assignment/views/assignment-edit___het-ophalen-aanmaken-van-de-opdracht-is-mislukt'
+				),
+				icon: IconName.alertTriangle,
+			});
+		}
+		setAssigmentLoading(false);
+	}, [user, match.params.id, tText, history, setAssignment]);
+
+	// Fetch initial data
+	useEffect(() => {
+		fetchAssignment();
+	}, [fetchAssignment]);
+
+	// Render
+
+	const renderHeaderButtons = () => {
+		return (
+			<ButtonToolbar>
+				<Spacer margin="left-small">
+					<Button
+						type="primary"
+						icon={IconName.edit}
+						label={tText('assignment/views/assignment-response-edit___bewerken')}
+						title={tText(
+							'assignment/views/assignment-response-edit___pas-deze-opdracht-aan'
+						)}
+						onClick={() =>
+							history.replace(
+								generatePath(APP_PATH.ASSIGNMENT_EDIT.route, {
+									id,
+								})
+							)
+						}
+					/>
```

Comment:
its better to wrap the button in a link, so meemoo can use right click => open in new tab on all buttons
```suggestion
					<Link to={generatePath(APP_PATH.ASSIGNMENT_EDIT.route, {
									id,
								})}>
						<Button
							type="primary"
							icon={IconName.edit}
							label={tText('assignment/views/assignment-response-edit___bewerken')}
							title={tText(
								'assignment/views/assignment-response-edit___pas-deze-opdracht-aan'
							)}
						/>
					</Link>
```

This was a specific feature request of meemoo in the past, so they expact this to be implemented the same way for future links/buttons